### ISSUE TITLE
Template variable not updating

### ISSUE THREAD

**Issue by hafeja** [grafana/grafana-json-datasource#95]
Hello @marcusolsson, first of all: Useful plugin, thanks!

I encountered an issue.

**Scenario**
The JSON plugin is used to define a template variable `json-var`. Inside the "Fields" tab, another template variable `var` (e.g. of type 
"Custom") is used: `$.mapping[?(@.attr == '$var')].something`.
![image](https://user-images.githubusercontent.com/59864250/114670987-df322400-9d03-11eb-8813-5a5b539473ae.png)

**Expected behavior**
`json-var` is always updated when `var` is updated.

**Actual behavior**
`json-var` isn't updated when `var` is updated.
Only once during creation of `json-var` it is initialized correctly according to `var`.

**Environment**
Grafana 7.5
Neither changing "Refresh" option nor "Cache Time" solves the issue.

Basically, I would like to use a template variable inside the "Field" configuration of the "Fields" tab. How can I get it working?

**Comment by marcusolsson** [comment]
Thank you for the excellent bug report. Much appreciated! 

This seems to be a duplicate of https://github.com/marcusolsson/grafana-json-datasource/issues/93. Could you confirm?

**Comment by hafeja** [comment]
From user perspective, this issue is related to usage of variables inside the "Field" configuration, not to the cache. But I don't know the internals of this plugin, maybe it's the same root cause.

If this issue disappears after #93 is fixed, then yes - it is a duplicate, otherwise not :smirk:

**Comment by marcusolsson** [comment]
I've just published [v1.1.0](https://grafana.com/grafana/plugins/marcusolsson-json-datasource/) to the marketplace. While I haven't worked on this specifically. I've fixed a couple of related issues. Would you mind letting me know if this issue is still there in 1.1.0?

**Comment by hafeja** [comment]
Unfortunately, I can still reproduce the issue :worried:

One further thing which might help to find the root cause: Inside the Settings menu of the Template variable `json-var` (from my example above) , the variable `var` is replaced correctly by its actual value. That means the "Preview of values" section gets updated correctly. But as soon as I save the setting and go back to the dashboard, `json-var` is not updated anymore.

**Comment by marcusolsson** [comment]
Thank you for confirming. I'll do some more digging!

**Comment by marcusolsson** [comment]
I've been experimenting with this, and while I'm able to reproduce this, I'm not sure this is something I can control in the plugin. Grafana isn't calling the function (`metricFindQuery`) to update the values at all, which makes me wonder whether this is something Grafana supports. 

Are you able to do this with any other data source?

**Comment by marcusolsson** [comment]
So to clarify, if you've set `json-var` to refresh on Dashboard Load, it will indeed use the value set by `var` when the dashboard loads. So the question is whether Grafana lets you refresh when the variable it depends on change. I've tried to reproduce this with other data sources, but have been unable to do so. 

Could you give me some hope that this is actually supported?

**Comment by hafeja** [comment]
A rather common use case for me is to have a chain of depending dashboard variables like this:

`var-a` depends on `var-b` depends on `var-c`.

The variables are defined using MySQL as datasource. If `var-c` is updated, the depending variables `var-b` and `var-a` are updated properly. That's why I assume Grafana itself supports this use case.

**Comment by marcusolsson** [comment]
Thanks for confirming. I've also received confirmation from the team that this should indeed work. I'll do some more troubleshooting to see why it's not triggering.

**Comment by ugoleleux** [comment]
@marcusolsson https://github.com/marcusolsson/grafana-json-datasource/issues/216#issuecomment-992540623
I juste updated the closed issue, but here is my entry again on this case :
I found a new behavior that will help you in your analysis.
I have a variable $name that contains a list of choices.
This variable is used inside a child variable on the Params tab :
![image](https://user-images.githubusercontent.com/95695567/145833233-d65852e2-decd-4e81-ba60-b42e93ebef4d.png)

When I look at the dependency tree, the $name var is not used by $child
![image](https://user-images.githubusercontent.com/95695567/145833089-5c4e15e3-a5d8-42fc-bba2-f5c207cf2931.png)

If I refer $name variable in the Path tab :
![image](https://user-images.githubusercontent.com/95695567/145833126-6888be88-7e42-4419-964e-01d9b37cdb86.png)

This time, the dependency tree is well seeing the relation.
![image](https://user-images.githubusercontent.com/95695567/145833194-d052ba5e-0741-4985-9de0-2caba1bf2f78.png)


By doing this little change, my child var is now updated on the fly when I modify the $name.

I hope this will help you to correct this issue :)

Regards
Ugo

**Comment by bode689** [comment]
Hi all,
I just realized that _parent-var_ does update correctly if it is referenced inside the body of _child-var_.
Unfortunately, if the query of _child-var_ is a GET request and we need to pass _parent-var_ as a Param , it updates only after a manual refresh.

Hope it helps

**Comment by mlcdf** [comment]
@ugoleleux : not all heroes wear capes. Thank you!

**Comment by ugoleleux** [comment]
> @ugoleleux: not all heroes wear capes. Thank you for the workaround, it works great!

I'm glad it helped you, I spent so much time discovering this workaround. :)
